cambridgeScript/interpreter/variables.py

```diff
@@ -2,11 +2,11 @@
 
 from cambridgeScript.parser.lexer import Value
 from cambridgeScript.syntax_tree import FunctionDecl, ProcedureDecl
-
+from cambridgeScript.syntax_tree.types import Type
 
 @dataclass
 class VariableState:
-    variables: dict[str, Value | None] = field(default_factory=dict)
+    variables: dict[str, (list[Value] | Value | None, Type)] = field(default_factory=dict)
```

Comment:
`tuple[]` is the correct type-hint for tuples.

```suggestion
    variables: dict[str, tuple[list[Value] | Value | None, Type]] = field(default_factory=dict)
```